src/ui/component/checkbox/checkbox.tsx

```diff
@@ -6,22 +6,22 @@ import { Icon } from '@/ui/component/icon/icon'
 import './checkbox.scss'
 
 export type CheckboxOption = {
-  checked: boolean
   value: string
+  checked?: boolean
   id?: string
-  name?: string
+  label?: string
```

Comment:
Input type checkbox has a `name` attribute but no `label` (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox).
Checkbox component renders only the checkbox input, and DynamicCheckbox renders the `Checkbox` component with an highlighted label.
That's why I recommend to keep the name attribute for both Checkbox and DynamicCheckbox components.

It might be good to rename `DynamicCheckbox` component in a dedicated PR to avoid confusion.